imap_processing/spice/geometry.py

```diff
@@ -443,3 +444,74 @@ def instrument_pointing(
     if isinstance(et, typing.Collection):
         return np.rad2deg([spice.reclat(vec)[1:] for vec in pointing])
     return np.rad2deg(spice.reclat(pointing)[1:])
+
+
+def cartesian_to_spherical(
+    v: NDArray,
+) -> NDArray:
+    """
+    Convert cartesian coordinates to spherical coordinates.
+
+    Parameters
+    ----------
+    v : np.ndarray
+        A NumPy array with shape (n, 3) where each
+        row represents a vector
+        with x, y, z-components.
+
+    Returns
+    -------
+    spherical_coords : np.ndarray
+        Spherical coordinate.
+    """
+    vx = v[:, 0]
+    vy = v[:, 1]
+    vz = v[:, 2]
```

Comment:
Is there a reason not to use the same indexing as you did in the `spherical_to_cartesian` function like:
```suggestion
    vx = v[..., 0]
    vy = v[..., 1]
    vz = v[..., 2]
```